keps/sig-apps/3850-backoff-limits-per-index-for-indexed-jobs/README.md

```diff
@@ -0,0 +1,1266 @@
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+#### Alpha
+
+- the feature implemented behind the `JobBackoffLimitPerIndex` feature flag
+- The `JobBackoffLimitPerIndex` feature flag disabled by default
+- Tests: unit and integration
+
+#### Beta
+
+- Address reviews and bug reports from Alpha users
+- E2e tests are in Testgrid and linked in KEP
+- The feature flag enabled by default
+
+#### GA
+
+- Address reviews and bug reports from Beta users
+- Write a blog post about the feature
+- Graduate e2e tests as conformance tests
+- Lock the `JobBackoffLimitPerIndex` feature gate
+- Declare deprecation of the `JobBackoffLimitPerIndex` feature gate in documentation
+
+### Upgrade / Downgrade Strategy
+
+#### Upgrade
+
+An upgrade to a version which supports this feature should not require any
+additional configuration changes. In order to use this feature after an upgrade
+users will need to configure their Jobs by specifying
+`.spec.runPolicy.backoffLimitPerIndex`.
+There is no difference in behavior of Jobs if `.spec.runPolicy.backoffLimitPerIndex` is
+not set.
+
+#### Downgrade
+
+A downgrade to a version which does not support this feature should not require
+any additional configuration changes. Jobs which specified
+`.spec.runPolicy.backoffLimitPerIndex` (to make use of this feature) will be
+handled in a default way, ie. using the `.spec.backoffLimit`.
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+### Version Skew Strategy
+
+This feature is limited to control plane.
```

Comment:
Right, but you still have coordination between the kube-apiserver and kube-controller-manager, don't you?

Reply:
Correct, but withing control plane we only need to support skew when kube-apiserver is N+1 relative to kube-controller-manger (https://kubernetes.io/releases/version-skew-policy/#kube-controller-manager-kube-scheduler-and-cloud-controller-manager). So, we can be sure that the Job controller in kube-controller-manager operates on Job objects which support the new API.

Reply:
Added a short note along the lines.